app/workers/immediate_email_generation_worker.rb

```diff
@@ -40,6 +42,19 @@ def update_content_change_cache(subscription_contents)
     content_changes.merge!(ContentChange.where(id: ids).index_by(&:id))
   end
 
+  def messages
+    @messages ||= {}
+  end
+
+  def update_message_cache(subscription_contents)
+    message_ids = subscription_contents.flat_map { |_, sc| sc.map(&:message_id) }
+                                       .compact
+                                       .uniq
+    ids = message_ids - messages.keys
```

Comment:
I don't really understand this line as here `messages` will always be an empty array at this point unless I'm misunderstanding something?